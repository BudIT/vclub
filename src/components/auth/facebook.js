import React, { Component, PropTypes } from 'react';

const FacebookLogin = () => {
  window.fbAsyncInit = () => {
    FB.init({
      appId: '157524328028332',
      cookie: true,
      xfbml: true,
      version: 'v2.8',
    });
  };



  function myFacebookLogin() {
    FB.getLoginStatus((response) => {
      if (response.status === 'connected') {
        console.log('Logged in.');
        FB.api('/me/feed', (response) => {
          console.log(JSON.stringify(response));
        });
      }
    });
  }

  return (
    <div id="fb-root">
      <button
        className="fb-login-button"
        data-max-rows="1"
        data-size="medium"
        data-show-faces="false"
        data-auto-logout-link="true"
        onClick={myFacebookLogin}
      />
    </div>
  );
};

(function (d, s, id) {
  const fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) return;
  const js = d.createElement(s); js.id = id;
  js.src = '//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.8&appId=157524328028332';
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));

export default FacebookLogin;

