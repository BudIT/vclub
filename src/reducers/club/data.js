import createReducer from 'vclub/redux/createReducer';

import { auth } from 'vclub/actions/core';


export default createReducer({
  [auth]: (atom) => ({
    ...atom,
    authenticating: true,
  }),
}, {
  authenticated: false,
  authenticating: false,
  currentRoom: 'chat',
});
