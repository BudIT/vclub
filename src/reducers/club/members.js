import createReducer from 'vclub/redux/createReducer';

import { memberEnter, memberLeave } from 'vclub/actions/core';


export default createReducer({
  [memberEnter]: (atom, member) => [member, ...atom],
  [memberLeave]: (atom, memberId) => atom.filter((member) => member.id !== memberId),
}, []);
