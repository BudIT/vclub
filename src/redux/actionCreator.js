export default function actionCreator(type, enhancers) {
  function enhance(action, args) {
    if (enhancers) {
      enhancers.forEach((e) => e(action, ...args));
    }
  }

  function actionCreatorFn(...args) {
    const action = {
      type,
      payload: args[0],
      meta: {},
    };

    enhance(action, args);

    if (action.payload instanceof Error) {
      action.error = true;
    }

    return action;
  }

  actionCreatorFn.toString = () => type;
  actionCreatorFn.displayName = type;

  return actionCreatorFn;
}
