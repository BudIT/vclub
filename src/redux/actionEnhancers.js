export function setPayload(fn) {
  return (action, ...args) => {
    // eslint-disable-next-line no-param-reassign
    action.payload = fn(...args);
  };
}

export function setMeta(field, fn) {
  return (action, ...args) => {
    // eslint-disable-next-line no-param-reassign
    action.meta[field] = fn(...args);
  };
}

export function setMetaValue(field, value) {
  return (action) => {
    // eslint-disable-next-line no-param-reassign
    action.meta[field] = value;
  };
}

export function withSideEffect(sideEffect) {
  return setMetaValue('sideEffect', sideEffect);
}

export const broadcast = setMetaValue('broadcast', true);
export const remote = setMetaValue('remote', true);
