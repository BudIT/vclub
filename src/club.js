import React from 'react';
import io from 'socket.io-client';
import { render } from 'react-dom';
import { createStore, compose, applyMiddleware } from 'redux';

import { Provider } from 'react-redux';

import ClubLayout from 'vclub/views/clubLayout/ClubLayout';

import clientActionBroker from 'vclub/redux/middlewares/clientActionBroker';
import sideEffectProcessor from 'vclub/redux/middlewares/sideEffectProcessor';
import reducer from 'vclub/redux/clubReducer';
import initialState from 'vclub/redux/initialClubState';
import { restoreAuth } from 'vclub/redux/club/auth';


const ioSocket = io({ path: '/vclub-socket', forceNew: true });

const storeEnhancer = compose(
  applyMiddleware(
    clientActionBroker(ioSocket),
    sideEffectProcessor({ context: { ioSocket, localStorage } }),
  ),
  window.devToolsExtension ? window.devToolsExtension() : f => f
);

const store = createStore(reducer, initialState, storeEnhancer);

ioSocket.on('dispatch', action => store.dispatch(action));

store.dispatch(restoreAuth());

render((
  <Provider store={store}>
    <ClubLayout />
  </Provider>
), document.getElementById('AppRoot'));
