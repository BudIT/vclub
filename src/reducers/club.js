import { combineReducers } from 'redux';
import { INITIALIZE } from 'vclub/constants/actionTypes';

import membersReducer from './club/members';
import dataReducer from './club/data';


const baseReducer = combineReducers({
  members: membersReducer,
  data: dataReducer,
});

export default function clubReducer(atom, action) {
  if (action.type === INITIALIZE) {
    const newAtom = action.payload;
    newAtom.data.authenticated = true;
    newAtom.data.authenticating = false;

    return newAtom;
  }

  return baseReducer(atom, action);
}
