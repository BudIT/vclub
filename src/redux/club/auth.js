import actionCreator from 'borex-actions/actionCreator';
import commandCreator from 'borex-actions/commandCreator';
import withSideEffect from 'borex-actions/withSideEffect';
import createReducer from 'borex-reducers/createReducer';
import setIn from 'borex-reducers/setIn';


export const auth = actionCreator(
  withSideEffect((context, authData, remember = true) => {
    const { ioSocket, localStorage, Raven } = context;

    ioSocket.emit('auth', authData);

    Raven.setUserContext(authData);

    Raven.captureMessage('Authenticated', {
      level: 'info',
      logger: 'auth',
      extra: authData,
    });

    if (remember) {
      localStorage.setItem('storedAuth', JSON.stringify(authData));
    }
  }),
);

export const setRestoredData = actionCreator();

export const restoreAuth = commandCreator((context) => {
  const { dispatch, localStorage } = context;
  const authJSON = localStorage.getItem('storedAuth');

  if (authJSON) {
    dispatch(setRestoredData(JSON.parse(authJSON)));
  }
});

export const logOut = commandCreator(() => {
  document.location.reload();
});


export default createReducer(on => {
  on(auth, setIn('authenticating', () => true));
  on(setRestoredData, setIn('restored'));
});
