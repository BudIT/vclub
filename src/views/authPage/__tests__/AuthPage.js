/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */

import React from 'react';

import { reducer as formReducer } from 'redux-form';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { auth } from 'vclub/redux/club/auth';

import AuthPageContainer, { AuthPageComponent } from '../AuthPage';

function createPage(subject) {
  return {
    usernameField: () => subject.find('.input_name'),
    form: () => subject.find('.login'),
    errorBlock: () => subject.find('.errors'),
  };
}

const setTextToInputThenBlur = (page, text) => {
  const input = page.usernameField();
  input.simulate('focus');
  input.node.value = text;
  input.simulate('blur');
};

describe('<AuthPage /> container', () => {
  let store;
  let subject;
  let page;
  const dispatchSpy = jest.fn();

  beforeEach(() => {
    store = createStore(combineReducers({ form: formReducer }));
    subject = mount(
      <Provider store={store}>
        <AuthPageContainer dispatch={dispatchSpy} />
      </Provider>
    );
    page = createPage(subject);
  });

  it('shows help text when first name is set to blank', () => {
    setTextToInputThenBlur(page, '');
    expect(page.errorBlock().text()).toBe('Пожалуйста, введите Ваше имя !!!');
  });

  it('shows help text when username is too long', () => {
    setTextToInputThenBlur(page, 'DenisViraBogdanYanisVeryVeryLongName');
    expect(page.errorBlock().text()).toBe('Имя должно быть 15 символов или меньше');
  });

  it('shows nothing when username field is correctly filled', () => {
    setTextToInputThenBlur(page, 'Zeus');
    expect(page.errorBlock().length).toBe(0);
  });

  it('calls onSubmit when the form is filled correctly', () => {
    const expectedAction = auth();

    setTextToInputThenBlur(page, 'Zeus');
    page.form().simulate('submit');
    expect(dispatchSpy).toHaveBeenCalledTimes(1);
    expect(dispatchSpy.mock.calls[0][0].type).toBe(expectedAction.type);
  });
});

test('<AuthPage /> component renders correctly', () => {
  // const store = createStore(() => ({}));

  const wrapper = shallow(
    <AuthPageComponent handleSubmit={jest.fn()} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
