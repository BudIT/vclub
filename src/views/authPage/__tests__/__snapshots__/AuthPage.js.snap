exports[`test <AuthPage /> component renders correctly 1`] = `
<section
  className="auth">
  <form
    className="login"
    onSubmit={[Function]}>
    <fieldset
      className="form_group">
      <div>
        <div>
          <Field
            className="input_name"
            component={[Function]}
            name="username"
            onFocus={true}
            placeholder="Имя..."
            type="text" />
        </div>
      </div>
      <div
        className="check_container">
        <div
          className="round_checkbox">
          <Field
            component="input"
            id="master"
            name="master"
            type="checkbox" />
          <label
            htmlFor="master">
            <h3
              className="label_h3">
              Ведущий
            </h3>
          </label>
        </div>
        <div
          className="round_checkbox">
          <Field
            component="input"
            id="remember"
            name="remember"
            type="checkbox" />
          <label
            htmlFor="remember">
            <h3
              className="label_h3">
              Запомнить
            </h3>
          </label>
        </div>
      </div>
      <button
        className="btn"
        type="submit">
        Войти
      </button>
    </fieldset>
  </form>
</section>
`;
